### PR TITLE
Release 0.9.0 docs checklist and cleanup

### PR DESCRIPTION
**Pull request by unityjeffrey**
Multiple commits to clean up documentation for new features and fixes.

Still a few outstanding issues:
~~- Linking to profiling in the release notes~~
- How does one enable the GAIL reward signal in hyperparameters?  Potentially more details needed in Reward-Signal.md

### CODE REVIEW

File: docs/ML-Agents-Overview.md [hunk 1]
```diff
@@ -319,11 +319,11 @@ imitation learning algorithm will then use these pairs of observations and
 actions from the human player to learn a policy. [Video
 Link](https://youtu.be/kpb8ZkMBFYs).
 
-ML-Agents provides ways to both learn directly from demonstrations as well as
-use demonstrations to help speed up reward-based training, and two algorithms to do
-so (Generative Adversarial Imitation Learning and Behavioral Cloning). The
-[Training with Imitation Learning](Training-Imitation-Learning.md) tutorial
-covers these features in more depth.
+The toolkit provides a way for agents to learn directly from demonstrations using
```
Comment (ervteng) [inline docs/ML-Agents-Overview.md]: GAIL can be used for both learning directly from demonstrations AND speeding up RL training. BC can technically do both as well (though we call the BC to speed up "PreTraining").

File: docs/Training-Imitation-Learning.md [hunk 2]
```diff
@@ -12,19 +12,21 @@ from a demonstration to learn a policy. [Video Link](https://youtu.be/kpb8ZkMBFY
...
-it is easier to just show the agent how to achieve the reward. In these cases,
-imitation learning can dramatically reduce the time it takes to solve the environment.
+the reward and thus not learn from it. Curiosity (which is available in the toolkit) 
+helps the agent explore, but in some cases
+it is easier to show the agent how to achieve the reward. In these cases,
+imitation learning combined with reinforcement learning can dramatically 
+reduce the time the agent takes to solve the environment.
 For instance, on the [Pyramids environment](Learning-Environment-Examples.md#pyramids),
-just 6 episodes of demonstrations can reduce training steps by more than 4 times.
+using 6 episodes of demonstrations (recorded from a human player) can reduce training steps by more than 4 times. See PreTraining + GAIL + Curiosity + RL below.
```
Comment (ervteng) [inline docs/Training-Imitation-Learning.md]: Not recorded from a human player for this plot.

File: docs/Training-PPO.md [hunk 3]
```diff
@@ -42,10 +42,10 @@ rewarding the agent for various different behaviors. For instance, we could rewa
 the agent for exploring new states, rather than just when an explicit reward is given.
 Furthermore, we could mix reward signals to help the learning process.
 
-`reward_signals` provides a section to define [reward signals.](Training-RewardSignals.md)
-ML-Agents provides two reward signals by default, the Extrinsic (environment) reward, and the
+Using `reward_signals` allows you to define [reward signals.](Reward-Signals.md)
+The ML-Agents toolkit provides two reward signals by default, the Extrinsic (environment) reward and the
```
Comment (ervteng) [inline docs/Training-PPO.md]: Three reward signals, Extrinsic, Curiosity, and GAIL
